haystack/document_stores/graphdb.py

```diff
@@ -49,37 +47,60 @@ def __init__(
+        :param timeout: How many seconds to wait for the server to send data before giving up,
+            as a float, or a :ref:`(connect timeout, read timeout) <timeouts>` tuple.
+            Defaults to 10 seconds.
+        :type timeout: Union[float, Tuple[float, float]]
         https://graphdb.ontotext.com/documentation/free/configuring-a-repository.html#configure-a-repository-programmatically
         """
         url = f"{self.url}/rest/repositories"
         files = {"config": open(config_path, "r", encoding="utf-8")}
-        response = requests.post(url, files=files, headers=headers)  # type: ignore
+        response = requests.post(url, files=files, headers=headers, timeout=timeout)  # type: ignore
         if response.status_code > 299:
             raise Exception(response.text)
 
-    def delete_index(self, headers: Optional[Dict[str, str]] = None):
+    def delete_index(self, headers: Optional[Dict[str, str]] = None, timeout: Union[float, Tuple[float, float]] = 10.0):
         """
         Delete the index that GraphDBKnowledgeGraph is connected to. This method deletes all data stored in the index.
         :param headers: Custom HTTP headers to pass to http client (e.g. {'Authorization': 'Basic YWRtaW46cm9vdA=='})
+        :param timeout: How many seconds to wait for the server to send data before giving up,
+            as a float, or a :ref:`(connect timeout, read timeout) <timeouts>` tuple.
+            Defaults to 10 seconds.
+        :type timeout: Union[float, Tuple[float, float]]
         """
         url = f"{self.url}/rest/repositories/{self.index}"
-        response = requests.delete(url, headers=headers)
+        response = requests.delete(url, headers=headers, timeout=timeout)
         if response.status_code > 299:
             raise Exception(response.text)
 
-    def import_from_ttl_file(self, index: str, path: Path, headers: Optional[Dict[str, str]] = None):
+    def import_from_ttl_file(
+        self,
+        index: str,
+        path: Path,
+        headers: Optional[Dict[str, str]] = None,
+        timeout: Union[float, Tuple[float, float]] = 10.0,
+    ):
         """
         Load an existing knowledge graph represented in the form of triples of subject, predicate, and object from a .ttl file into an index of GraphDB
 
         :param index: name of the index (also called repository) in the GraphDB instance where the imported triples shall be stored
         :param path: path to a .ttl containing a knowledge graph
         :param headers: Custom HTTP headers to pass to http client (e.g. {'Authorization': 'Basic YWRtaW46cm9vdA=='})
+        :param timeout: How many seconds to wait for the server to send data before giving up,
+            as a float, or a :ref:`(connect timeout, read timeout) <timeouts>` tuple.
+            Defaults to 10 seconds.
+        :type timeout: Union[float, Tuple[float, float]]
```

Comment:
No need to add the type here as we already have it in the type annotation in the methods signature